### PR TITLE
Support JWT authentication

### LINKED ISSUE #59: implement 'access-token' option
trino cli has `access-token` option to passing token. if access-token set every request has token info.

### PR DESCRIPTION
**Pull request by nineinchnick**
Based on #110.

I added certificate generation to be able to run integration tests. Statically generated certificates cannot be used, since on macOS they could only be valid for 1 year, which would require regenerating them.

I also refactored `trino/integration_tls_test.go` to avoid using the reverse proxy.

Fixes #59

### CODE REVIEW

**Comment by nineinchnick** [comment]
@losipiuk @wendigo PTAL, there are a lot of devex related changes here, but I made sure every commit contain single logical change.

**Comment by wendigo** [comment]
@nineinchnick please rebase

**Comment by nineinchnick** [comment]
@wendigo good to go, the one failure is a flaky test. I'll see if I can adress that later

**Comment by wendigo** [comment]
@nineinchnick please rebase, there is a conflict

**Comment by nineinchnick** [comment]
@wendigo green again

**Comment by mosabua** [comment]
Thanks for running with this @nineinchnick

**Comment by nineinchnick** [comment]
My main motivation is how quickly I can get reviews in this repo :D

**Comment by nineinchnick** [comment]
@wendigo PTAL

File: trino/etc/config.properties [hunk 1]
```diff
@@ -4,3 +4,13 @@ node-scheduler.include-coordinator=true
 http-server.http.port=8080
 discovery-server.enabled=true
 discovery.uri=http://localhost:8080
+
```
Comment (wendigo) [inline trino/etc/config.properties:7]: remove these blank lines - one is enough

File: trino/integration_test.go [hunk 2]
```diff
@@ -120,6 +143,104 @@ func TestMain(m *testing.M) {
...
+		return fmt.Errorf("failed to generate private key: %w", err)
+	}
+
+	serialNumberLimit := new(big.Int).Lsh(big.NewInt(1), 128)
+	serialNumber, err := rand.Int(rand.Reader, serialNumberLimit)
+	if err != nil {
+		return fmt.Errorf("failed to generate serial number: %w", err)
+	}
+
+	template := x509.Certificate{
```
Comment (wendigo) [inline trino/integration_test.go:168]: neat!

**Review by wendigo — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -11,7 +11,7 @@ Trino, and receive the resulting data.
 
 * Native Go implementation
 * Connections over HTTP or HTTPS
-* HTTP Basic and Kerberos authentication
+* HTTP Basic, and Kerberos authentication
```
[the remark this answers is not in the document]
Comment (nineinchnick) [reply README.md]: The Oxford comma?

File: trino/etc/config.properties (hunk 1 above)
Comment (nineinchnick) [reply trino/etc/config.properties:7]: Done

**Pull request merged**